Mathlib/Data/List/Count.lean

```diff
@@ -181,6 +181,16 @@ theorem prod_eq_pow_single [Monoid α] (a : α)
 #align list.prod_eq_pow_single List.prod_eq_pow_single
 #align list.sum_eq_nsmul_single List.sum_eq_nsmul_single
 
+lemma count_filter_add_count_filter {α} [DecidableEq α] (P : α → Prop) [DecidablePred P]
```

Comment:
This is a special case of `List.countP_eq_countP_filter_add` proved elsewhere.

Reply:
OK; this is in `Mathlib.Data.List.Perm`, because it uses `Perm.countP_eq` in the proof. (Which is exactly the opposite way of how I was dealing with these lemmas :smile:)
Would it still make sense to add a `count` (as opposed to `countP`) version (and then in `Mathlib.Data.List.Perm`, to be able to use the existing lemma), or is it too close?

Reply:
I think the lemma statement is fine, but it should be proved using the fact that `l.filter P ++ l.filter (¬ P ·) ~ l` and `count a` distributes over `perm`

Reply:
The problem with that is that `List.Perm` is not visible in `Data.List.Count`. So again, the lemma would have to go to `Data.List.Perm`. Ist this what you recommend?

Reply:
Moved now (and swapped the sides of the equality, changed the name accordingly) and proved using `List.countP_eq_countP_filter_add`.